presto-main/src/main/java/com/facebook/presto/execution/scheduler/ScaledWriterScheduler.java

```diff
@@ -105,15 +116,30 @@ private int getNewTaskCount()
             return 1;
         }
 
+        if (optimizedScaleWriterProducerBuffer) {
+            double totalProducerBufferUtilization = sourceTasksProvider.get().stream()
```

Comment:
I'm a little bit confused about this loging. I was thinking more about changing the condition to something like `if 50% overutilized OR 90% non empty - add more writer`. Thoughts?

Reply:
@arhimondr : Addressed the comments per offline discussion. Basically now we add more writers if the overall buffer utilization is more than the number of writers. 

We don't need to worry about accidentally open too many writers since overall `writtenBytes` has to be more than `writerMinSizeBytes * scheduledNodes.size()`

```
                    writtenBytes >= writerMinSizeBytes * scheduledNodes.size()) 
```